applications/GeoMechanicsApplication/tests/cpp_tests/test_T_micro_climate_flux_condition.cpp

```diff
@@ -0,0 +1,437 @@
+                                  const Variable<double>& rVariable)
+{
+    for (auto& node : rNodes) {
+        node.FastGetSolutionStepValue(rVariable, 0) = 1.0;
+        node.FastGetSolutionStepValue(rVariable, 1) = 0.0;
+    }
+}
+
+void AddSolutionStepValuesToNodes(ModelPart::NodesContainerType& rNodes,
+                                  const std::vector<const Variable<double>*>& rVariables)
+{
+    for (auto p_variable : rVariables) {
+        AddSolutionStepValuesToNodes(rNodes, *p_variable);
+    }
+}
+
+ModelPart& CreateDummyModelPartWithNodes(Model& rModel,
+                                         const std::function<void(ModelPart&)>& rCreateNodesFunc)
+{
+    constexpr auto buffer_size = Model::IndexType{2};
+    auto& r_result = rModel.CreateModelPart("dummy", buffer_size);
+
+    const auto variables = std::vector<const Variable<double>*>{&AIR_TEMPERATURE,
+                                                                &AIR_HUMIDITY,
+                                                                &SOLAR_RADIATION,
+                                                                &WIND_SPEED,
+                                                                &PRECIPITATION,
+                                                                &TEMPERATURE,
+                                                                &DT_TEMPERATURE};
+    AddSolutionStepVariablesToModelPart(r_result, variables);
+
+    rCreateNodesFunc(r_result);
+    AddSolutionStepValuesToNodes(r_result.Nodes(), variables);
+
+    r_result.GetProcessInfo()[DELTA_TIME] = 0.5;
+
+    return r_result;
+}
+
+intrusive_ptr<Condition> CreateMicroClimateCondition(ModelPart& rModelPart,
+                                                     shared_ptr<Properties> pProperties,
+                                                     std::size_t DimensionSize)
+{
+    constexpr auto condition_id = std::size_t{1};
+    auto node_ids = std::vector<ModelPart::IndexType>{};
+    for (const auto& node : rModelPart.Nodes()) {
+        node_ids.emplace_back(node.Id());
+    }
+    const auto condition_name = std::string{"GeoTMicroClimateFluxCondition"}
+                                    + std::to_string(DimensionSize) + "D"
+                                    + std::to_string(node_ids.size()) + "N";
+    return rModelPart.CreateNewCondition(condition_name, condition_id, node_ids, pProperties);
+}
+
+std::string ExecuteInitializeSolutionStep(intrusive_ptr<Condition> pCondition,
+                                          const ProcessInfo& rProcessInfo)
+{
+    try {
+        pCondition->InitializeSolutionStep(rProcessInfo);
+    }
+    catch (const Exception& e) {
+        return e.what();
+    }
+
+    return {}; // no error message
+}
+
+constexpr auto relative_tolerance = 1.0e-3;
+constexpr auto absolute_tolerance = 1.0e-3;
+
+}
+
+
+namespace Kratos::Testing {
+
+KRATOS_TEST_CASE_IN_SUITE(NoThrowWhenInitializingThermalMicroClimateCondition, KratosGeoMechanicsFastSuite)
```

Comment:
Due to the great generalization, the tests look almost the same. Is it possible to combine them? Or will it be done later with gtest?

Reply:
The scope of the tests is still limited and in this way they are easy to read independently, so for now I'd propose to keep this as it is.